pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/RebalanceCommand.java

```diff
@@ -0,0 +1,114 @@
+import org.kohsuke.args4j.Option;
+
+
+/**
+ * A sub-command for pinot-admin tool to rebalance a specific table
+ */
+public class RebalanceCommand extends AbstractBaseAdminCommand implements Command {
+
+  @Option(name = "-zkAddress", required = true, metaVar = "<http>", usage = "HTTP address of Zookeeper.")
+  private String _zkAddress;
+
+  @Option(name = "-clusterName", required = true, metaVar = "<String>", usage = "Pinot cluster name.")
+  private String _clusterName;
+
+  @Option(name = "-tableName", required = true, metaVar = "<String>", usage = "Name of the table to rebalance (e.g. myTable_OFFLINE)")
+  private String _tableName;
+
+  @Option(name = "-tableType", required = true, metaVar = "<String>", usage = "Type of table (OFFLINE or REALTIME)")
+  private String _tableType;
+
+  @Option(name = "-dryRun", required = false, metaVar = "<boolean>", usage = "Dry Run (just get the target ideal state without rebalancing)")
+  private boolean _dryRun;
+
+  @Option(name = "-includeConsuming", required = false, metaVar = "<boolean>", usage = "Should include consuming segments? Applicable only for realtime tables")
+  private boolean _includeConsuming;
+
+  @Option(name = "-noDowntime", required = false, metaVar = "<boolean>", usage = "Rebalance without downtime?")
+  private boolean _noDowntime;
+
+  @Option(name = "-minReplicasToKeepUpForNoDowntime", required = false, metaVar = "<int>", usage = "Minimum number of serving replicas that will be kept up for no downtime rebalancing")
+  private int _minReplicasForNoDowntime;
+
+  @Option(name = "-help", required = false, help = true, aliases = {"-h", "--h", "--help"}, usage = "Print this message.")
+  private boolean _help;
+
+  public boolean getHelp() {
+    return _help;
+  }
+
+  @Override
+  public String getName() {
+    return "rebalanceTable";
+  }
+
+  @Override
+  public boolean execute()
+      throws Exception {
+    final PinotTableRebalancer tableRebalancer =
+        new PinotTableRebalancer(_zkAddress, _clusterName, _dryRun, _noDowntime, _includeConsuming,
+            _minReplicasForNoDowntime);
+    return tableRebalancer.rebalance(_tableName, _tableType);
+  }
+
+  @Override
+  public String description() {
+    final String description =
+        "Rebalances segments of a table. " + "By default, rebalancing is done with downtime where there is no guarantee"
+            + "of any existing serving replica being alive for serving queries. "
+            + "Use the -nodowntime option to rebalance without downtime. It takes"
+            + "care of keeping at least 1 serving replica alive while rebalancing. "
+            + "For better QPS, use the -minReplicasForNoDowntime <val> option along with"
+            + "-nodowntime to ensure that rebalancer keeps at least the required number of "
+            + "replicas alive for each segment while segments are being moved between hosts";
+    return description;
+  }
+
+  @Override
+  public void printUsage() {
```

Comment:
Why are you trying to override `printUsage`?

I thought the framework would deal with printing information correctly (with the info that you wrote in `@Option` annotation

Reply:
printUsage() is overriden to also print some example usage of RebalanceCommand. So I let it print each option along with description by calling super.printUsage() and then print some sample usages.

Reply:
Can the --help option serve here?

Reply:
No that doesn't print example usage.

Reply:
Can you change the super method to call a printExamples() and printUsage()? That way other commands can also take advantage of this to print examples.  No better documentation than the one printed by the code!

Reply:
Done